### ISSUE TITLE
Optimise or remove PaperTrail

### ISSUE THREAD

**Issue by alexandria-gfsc** [geeksforsocialchange/PlaceCal#1087]
https://github.com/geeksforsocialchange/PlaceCal/blob/main/app/models/calendar.rb#L55

This code is only used in one place and seems to show recently modified events (updated, deleted). PaperTrail is used specifically to catch deleted events without doing any specific (handrolled) work to archive those events.

However, it's causing a timeout on a few calendar pages in production and could be quite troublesome based on the SQL query it executes now we have more throughput from calendars. On most calendar pages it returns absolutely nothing, however on certain calendar pages it executes a huge SQL query that causes the page to timeout (Viewing the log I've captured from one of these entries in certain editors causes huge freezing and slowdown of the editor...).

I'm not sure how useful this is from a user perspective though -- if it is useful we should find a better way to store recently (within about 2 weeks) deleted and updated events for this purpose (maybe just have a flag on them, or better yet stick them in a separate table), so that the lookup can be cleaner and organized per-calendar.

TL;DR: PaperTrail is annoying and:

- Only used in one single function
- Returns a value that is used exclusively in view code
- For something that we could do without PaperTrail, seemingly, and a hell of a lot faster on the view code rendering I think (Would make event deletion slower however this _might_ be negligible).
- It's causing timeout on edit pages to the point that the site gives up

**Comment by GabAlexandria** [comment]
Papertrail was just a convenient way to quickly track creation and deletion of events at the time without building out the full infrastructure. It would definitely be better to roll your own solution has papertrail is known to have performance issues.

**Comment by alexandria-gfsc** [comment]
Given @kimadactyl's reticence in discord to remove PaperTrail here (and I don't entire disagree with her points), I did some digging and we don't seem to be setting `version_limit`. My guess is that PaperTrail is storing history of all events since the beginning of time with no limit

Adding a `version_limit` line to our config as in: https://github.com/paper-trail-gem/paper_trail#2e-limiting-the-number-of-versions-created
might be a possible alternative fix :)

Since we are literally only using it for Events, we could also constrict it to the Events class model and limit our total data size, too: https://github.com/paper-trail-gem/paper_trail#per-class

So it might boil down to a 2 line fix :+1:
